### PR TITLE
CASSANDRASC-50: Deprecate the sidecar cassandra health endpoint conta…

### PR DESCRIPTION
**Pull request by frankgh**
…ining instance segment

This commit deprecates the Cassandra Health endpoint containing the instance segment in the path. This endpoint is currently unused and it is replaced by the health endpoint with the `instanceId` query string parameter. Since the `instanceId` is optional we move the path param (mandatory) to the query param (optional).

This commit also moves the CassandraHealthService from jax RS to a vertx Handler. It also moves the HealthService to a inlined handler, simplifying the service.

### CODE REVIEW

**Review by dineshjoshi — APPROVED** [review]
+1

File: src/main/java/org/apache/cassandra/sidecar/cluster/InstancesConfig.java [hunk 1]
```diff
@@ -37,14 +37,16 @@
      *
      * @param id instance's id
      * @return instance meta information
+     * @throws IllegalArgumentException when the instance with {@code id} does not exist
      */
-    InstanceMetadata instanceFromId(final int id);
+    InstanceMetadata instanceFromId(int id) throws IllegalArgumentException;
```
Comment (yifan-c) [inline src/main/java/org/apache/cassandra/sidecar/cluster/InstancesConfig.java]: Does `NoSuchElementException` fit better?

**Review by yifan-c — APPROVED** [review]
LGTM with 1 nit

File: src/main/java/org/apache/cassandra/sidecar/cluster/InstancesConfig.java (hunk 1 above)
Comment (frankgh) [reply src/main/java/org/apache/cassandra/sidecar/cluster/InstancesConfig.java]: yeah, I think this makes more sense. I will update it

**Pull request closed without merging**